crates/compiler/mono/src/inc_dec.rs

```diff
@@ -779,7 +792,20 @@ fn insert_refcount_operations_stmt<'v, 'a>(
                         As the consumption should only ever increase.
                         Otherwise we will be looping forever."
                     );
+
+                    current_body_env
+                        .add_joinpoint_consumption(*joinpoint_id, joinpoint_consumption.clone());
+
+                    let all_empty_or_borrowed =
+                        current_body_env.is_joinpoint_consumptions_all_empty_or_borrowed();
+
+                    current_body_env.remove_joinpoint_consumption(*joinpoint_id);
+
                     joinpoint_consumption = current_joinpoint_consumption;
+
+                    if all_empty_or_borrowed {
+                        break (new_body, current_body_env);
+                    }
```

Comment:
After a lot of debugging I noticed we can just stop early in this case, which solved the perf problem.